src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageProcessorTest.java

```diff
@@ -0,0 +1,263 @@
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class NotificationMessageProcessorTest {
+
+
+    private NotificationMessageProcessor notificationMessageProcessor;
+
+    @Mock
+    private IMessageReceiver messageReceiver;
+    @Mock
+    private NotificationMessageHandler notificationMessageHandler;
+    @Mock
+    private NotificationMessageParser notificationMessageParser;
+
+    @BeforeEach
+    public void before() throws Exception {
+        notificationMessageProcessor = new NotificationMessageProcessor(
+            messageReceiver,
+            notificationMessageHandler,
+            notificationMessageParser,
+            5
+        );
+    }
+
+    @Test
+    public void should_return_true_when_there_is_a_message_to_process() throws Exception {
+        // given
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+        given(messageReceiver.receive()).willReturn(message);
+
+        NotificationMsg notificationMsg = mock(NotificationMsg.class);
+        given(notificationMessageParser.parse(messageBody)).willReturn(notificationMsg);
+        doNothing().when(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+
+        // then
+        assertThat(processedMessage).isTrue();
+        verify(messageReceiver).receive();
+        verify(notificationMessageParser).parse(messageBody);
+        verify(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+    }
+
+    @Test
+    public void should_return_false_when_there_is_no_message_to_process() throws Exception {
+        // given
+        given(messageReceiver.receive()).willReturn(null);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+        // then
+        assertThat(processedMessage).isFalse();
+    }
+
+    @Test
+    public void should_not_throw_exception_when_queue_message_is_invalid() throws Exception {
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+        given(messageReceiver.receive()).willReturn(message);
```

Comment:
Maybe it would make sense to extract such bit into a separate method? I think there's a few places that do this.

Reply:
I use the message body and message vars. I can make it  class level field

Reply:
Never mind, not so simple

Reply:
I changed add as class field.